compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -293,6 +293,11 @@ impl Diagnostic {
         self
     }
 
+    pub fn highlighted_help(&mut self, msg: Vec<(String, Style)>) -> &mut Self {
```

Comment:
Perhaps add a documentation comment describing why someone might want to use `highlighted_help` when implementing a diagnostic?

Reply:
Added docstring